Dimmer.h

```diff
@@ -1,73 +1,132 @@
-/*
- This is the Dimmer library to control AC loads, including dimmable lamps.
- 
- Copyright (c) 2015 Circuitar
- This software is released under the MIT license. See the attached LICENSE file for details.
+/** @file
+ * This is the Dimmer library to control the AC load power and also dimmerable lamps.
+ *
+ * Version 2.0 - Multi-instance objects.
+ *
+ * Copyright (c) 2015 Circuitar
+ * This software is released under the MIT license. See the attached LICENSE file for details.
  */
-
 #ifndef DIMMER_H
 #define DIMMER_H
 
 #include "Arduino.h"
-#include <TimerOne.h>
 
+/**
+ * Maximum number of triacs that can be used. This is also the default value.
+ */
 #define MAX_TRIAC 10
 
-#ifndef dimmerObject
-#define dimmerObject dimmer
-#endif
+/**
+ * NanoShield_zeroCross pin settings.
+ *
+ * @param zeroCrossPin	Change this parameter to the pin your zero cross is attached.
+ * @param zeroCrossInt	Change this parameter to the interruption correspondent to the pin of the zero cross. 
+ * @see https://www.arduino.cc/en/Reference/attachInterrupt for more information.
+ */
+#define zeroCrossPin    2
+#define zeroCrossInt    0
 
-class Dimmer{
-    
-public:
-    Dimmer();
-    void attachZeroCross(int zc_pin, int inter);
-    void attachTriac(byte pin);
-    int init();
-    int initRamp();
-    int initCount(byte resolution = 1);
+/**
+ * Possible operation modes for the dimmer library.
+ */
+#define NORMAL_MODE     0
+#define RAMP_MODE       1
+#define COUNT_MODE      2
+
+/**
+ * Possible light state for the dimmer library.
+ */
+#define OFF    0
+#define ON     1
+
+/**
+ * A ZeroCross Nanoshield or similar module to dimm AC lights.
+ */
+class DimmerControl
```

Comment:
We should try to make the class have the same name as the library and the .h. Why not use just "Dimmer"?